File: lib/logic/tools/crypto_and_encodings/chef_language/component.dart
// package interpreter;
import 'package:gc_wizard/logic/tools/crypto_and_encodings/chef_language/ingredient.dart';
/*
import interpreter.Ingredient.State;
*/
class Component {

  State _state;
  int value;

  Component(int n, State s) {
    value = n;
    _state = s;
  }

  Component.Contructor1(Ingredient ingredient) {
    value = ingredient.getAmount();
    _state = ingredient.getstate();
  }

  int getValue() {
    return value;
  }

  void setValue(int n) {
    value = n;
  }

  State getState() {
    return _state;
  }

  void setState(State s) {
    _state = s;
  }

  //@Override
  Component clone() {
    return Component(value, _state);
  }

  void liquefy() {
    _state = State.Liquid;
  }
}

File: lib/logic/tools/crypto_and_encodings/chef/component.java
package interpreter;

import interpreter.Ingredient.State;

public class Component {

	private Ingredient.State state;
	private int value;
	
	public Component(int n, Ingredient.State s) {
		value = n;
		state = s;
	}
	
	public Component(Ingredient ingredient) {
		value = ingredient.getAmount();
		state = ingredient.getstate();
	}

	public int getValue() {
		return value;
	}
	
	public void setValue(int n) {
		value = n;
	}
	
	public Ingredient.State getState() {
		return state;
	}
	
	public void setState(Ingredient.State s) {
		state = s;
	}
	
	@Override
	public Component clone() {
		return new Component(value, state);
	}

	public void liquefy() {
		state = State.Liquid;
	}
}